aten/src/ATen/native/Math.h

```diff
@@ -1279,15 +1382,106 @@ calc_i0e(T _x) {
   T x = std::abs(_x);
 
   if (x <= 8.0) {
-    auto A = chebyshev_coefficients_A<T>();
+    auto A = chebyshev_coefficients_i0e_A<T>();
     T y = (x / 2.0) - 2.0;
     return static_cast<T>(chbevl(y, A, 30));
   }
 
-  auto B = chebyshev_coefficients_B<T>();
+  auto B = chebyshev_coefficients_i0e_B<T>();
   return static_cast<T>(
       chbevl(static_cast<T>(32.0 / x - 2.0), B, 25) / std::sqrt(x));
 }
 
 // Upcast bfloat16 input to float for numerical accuracy purposes
 inline c10::BFloat16 calc_i0e(c10::BFloat16 a) { return calc_i0e(static_cast<float>(a)); }
+
+/*
+ * This function is derived from the implementation of the i0e function in the Cephes Math Library.
+ * See note [3-Clause BSD License for the Cephes Math Library].
+ *
+ * Computes an approximation of the first order modified Bessel function of the first kind.
+ * The approximation is actually two (sub)approximations, both using a Chebyshev polynomial expansion.
+ * One approximates the function over [0, 8], and the other over (8, infinity). This function takes the absolute value
+ * of all inputs to convert them into the domain of the approximation.
+ */
+template <typename T>
+static inline typename std::enable_if<std::is_floating_point<T>::value, T>::type
+calc_i1(T _x) {
+  T x = std::abs(_x);
+
+  if (x <= 8.0) {
+    const auto A = chebyshev_coefficients_i1e_A<T>();
+    T y = (x / 2.0) - 2.0;
+    const T out = static_cast<T>(std::exp(x) * x * chbevl(y, A, 29));
+    return (_x < 0.0) ? -out : out;
+  }
+
+  const auto B = chebyshev_coefficients_i1e_B<T>();
+  const auto out = static_cast<T>(
+      (std::exp(x) * chbevl(static_cast<T>(32.0 / x - 2.0), B, 25)) / std::sqrt(x));
+  return (_x < 0.0) ? -out : out;
+}
+
+template <>
+inline float calc_i1(float _x) {
+  using T = float;
+  T x = std::abs(_x);
+
+  if (x <= 8.0) {
+    const auto A = chebyshev_coefficients_i1e_A<T>();
+    T y = (x / 2.0) - 2.0;
+    const T out = static_cast<T>(std::exp(x) * x * chbevl(y, A, 17));
```

Comment:
If these values are the only two differences, maybe create a common function that accepts these as arguments? Then a template can specialize float and double calling the common helper with the desired values

Reply:
I tried using `std::array` but mobile and mac builds were failing.
I was planning to do some refactoring later. For now, I think we can return coefficients and their length as tuple.

Reply:
Done with `std::tuple` for now.